data_structures/hashing/quadratic_probing.py

```diff
@@ -12,6 +12,40 @@ def __init__(self, *args, **kwargs):
         super().__init__(*args, **kwargs)
 
     def _collision_resolution(self, key, data=None):
+        """
+        Quadratic probing is an open addressing scheme used for resolving
+        collisions in hash table.
+
+        It works by taking the original hash index and adding successive
+        values of an arbitrary quadratic polynomial until open slot is found.
+
+        Hash + 1², Hash + 2², Hash + 3² .... Hash + n²
+
+        e.g:
+        1. Create hash table with size 7
+        >>> qp = QuadraticProbing(7)
+        >>> qp.insert_data(90)
+        >>> qp.insert_data(340)
+        >>> qp.insert_data(24)
+        >>> qp.insert_data(45)
+        >>> qp.insert_data(99)
+        >>> qp.insert_data(73)
+        >>> qp.insert_data(7)
+        >>> qp.keys()
+        {11: 45, 14: 99, 7: 24, 0: 340, 5: 73, 6: 90, 8: 7}
+
+        2. Create hash table with size 8
+        >>> qp = QuadraticProbing(8)
+        >>> qp.insert_data(0)
+        >>> qp.insert_data(999)
+        >>> qp.insert_data(111)
+        >>> qp.keys()
+        {0: 0, 7: 999, 3: 111}
+
+        reference:
+            - https://en.wikipedia.org/wiki/Quadratic_probing
```

Comment:
Let's put the URL above the tests so visitors can look at the tests and implementation without any visual clutter.

Reply:
@cclauss I made the changes as mentioned by you